File: task-03/Haskell/Sub-Task2/code2.hs
import System.IO

main :: IO ()
main = do
    -- Open input.txt for reading
    inputHandle <- openFile "input.txt" ReadMode
    -- Read the entire content of input.txt and keep it in content
    content <- hGetContents inputHandle
    -- Close the input file
    hClose inputHandle

    -- Open output.txt for writing (creates the file if it doesn't exist, overwrites if it does)
    outputHandle <- openFile "output.txt" WriteMode
    -- Write the content to output.txt
    hPutStr outputHandle content
    -- Close the output file
    hClose outputHandle

    putStrLn "Content has been copied from input.txt to output.txt"

File: task-03/Elixir/Sub-Task2/code2.ex
defmodule FileCopy do
  def copy_file do
    case File.read("input.txt") do
      {:ok, content} ->
        case File.write("output.txt", content) do
          :ok -> IO.puts("Content copied from input.txt to output.txt")
          {:error, reason} -> IO.puts("Error writing to output.txt: #{reason}")
        end
      {:error, reason} ->
        IO.puts("Error reading input.txt: #{reason}")
    end
  end
end

FileCopy.copy_file()

